repos/tap/actions/definitions/local/upsertDefinitions.js

```diff
@@ -13,23 +13,25 @@ const { CATEGORIES } = Values
  *
  * @returns {void}
  */
-export const upsertDefinitions = (definitions=noOpObj, definitionTypes) => {
-  dispatch({
-    type: ActionTypes.UPSERT_ITEMS,
-    payload: {
-      category: CATEGORIES.DEFINITIONS,
-      items: definitions,
-    },
-  })
+export const upsertDefinitions = (definitions, definitionTypes) => {
+  definitions &&
+    dispatch({
+      type: ActionTypes.UPSERT_ITEMS,
+      payload: {
+        category: CATEGORIES.DEFINITIONS,
+        items: definitions,
+      },
+    })
 
   // Sort the definitions by type ( given | then | etc... )
   // This makes it easier to match to features when editing
-  dispatch({
-    type: ActionTypes.UPSERT_ITEMS,
-    payload: {
-      category: CATEGORIES.DEFINITION_TYPES,
-      items: definitionTypes || definitionsByType(definitions),
-    },
-  })
```

Comment:
as a reminder, you could also use the helper actions we have in `keg-core` to simplify these basic actions:

```js
import { upsertItems } from 'SVActions'
export const upsertDefinitions = (definitions, types) => {
  upsertItems(CATEGORIES.DEFINITIONS, definitions)
  upsertItems(CATEGORIES.DEFINITION_TYPES, definitionTypes || definitionsByType(definitions))
}
```

and the same thing for `setDefinitions`

Reply:
Forgot all about these actually, I'll update to use these

Reply:
updated